### PR TITLE
 Fix network list subgraph query

### PR DESCRIPTION
**Pull request by tilacog**
This PR updates the GraphQL Query and Schema used by the Oracle to query data from the Epoch Subgraph.

The new query introduced the possibility of the absence of the `GlobalState` entity, so the rest of the code was refactored to accommodate that.

Some minor changes were made under the `.local` directory to make the local environment work as expected.

### CODE REVIEW

File: crates/oracle/src/subgraph.rs [hunk 1]
```diff
@@ -207,3 +255,33 @@ where
...
+    {
+        let sorted = elements.into_iter().sorted();
+        let mut duplicates = HashSet::new();
+        for (a, b) in sorted.tuple_windows() {
+            if a == b {
+                duplicates.insert(a);
+            }
+        }
+        duplicates
+    }
```
Comment (neysofu) [inline crates/oracle/src/subgraph.rs:274]: There's https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.duplicates for this :smile:

File: crates/oracle/src/subgraph.rs [hunk 2]
```diff
@@ -46,18 +53,47 @@ impl From<&Config> for SubgraphQuery {
...
-            .await
-            .context("querying Epoch Subgraph")?;
-        let ResponseData { networks } = response;
-        Ok(networks)
+    type State = subgraph_state::SubgraphStateGlobalState;
+
+    async fn get_subgraph_state(&self) -> anyhow::Result<Option<Self::State>> {
+        Ok(query(self.url.clone()).await?)
+    }
+}
```
Comment (neysofu) [inline crates/oracle/src/subgraph.rs:61]: This is fine for now, as it's not within the scope of the PR, but I think `SubgraphApi` should get discarded and become a simple function instead.

**Review by neysofu — APPROVED** [review]

File: crates/oracle/src/subgraph.rs (hunk 1 above)
Comment (tilacog) [reply crates/oracle/src/subgraph.rs]: Thanks! Not my smartest move :smile: 
Fixed in 75148f3

File: crates/oracle/src/subgraph.rs (hunk 2 above)
Comment (tilacog) [reply crates/oracle/src/subgraph.rs:61]: The only reason this trait exist is to allow tests for the `SubgraphState` struct. 
Maybe we can think of some way to make it so without this trait.

**Pull request merged**